### PR TITLE
Remember user input in mlbench run

### LINKED ISSUE #56: Remember user input in mlbench run
When starting a custom run in mlbench, the user has to supply the image name, command and other parameters.

It'd be nice if the CLI could remember the last user input as default so the user doesn't have to keep inputting the same things over and over when starting runs.

### PR DESCRIPTION
**Pull request by mmilenkoski**
This resolves #56. I am not sure if pickle is the right format we want to use or if there is a better way.

### CODE REVIEW

File: mlbench_core/cli/cli.py [hunk 1]
```diff
@@ -113,6 +114,13 @@ def cli(args=None):
 @click.option("--dashboard-url", "-u", default=None, type=str)
 def run(name, num_workers, gpu, light, dashboard_url):
     """Start a new run for a benchmark image"""
+    current_run_inputs = {}
+
+    try:
+        last_run_inputs = pickle.load(open("last_run_inputs.pkl", "rb"))
+    except:
```
Comment (ehoelzl) [inline mlbench_core/cli/cli.py:121]: You need to catch an exception here. `except FileNotFoundError as e`.
Maybe also print a statement saying no previous inputs were found ?

File: mlbench_core/cli/cli.py [hunk 2]
```diff
@@ -113,6 +114,13 @@ def cli(args=None):
 @click.option("--dashboard-url", "-u", default=None, type=str)
 def run(name, num_workers, gpu, light, dashboard_url):
     """Start a new run for a benchmark image"""
+    current_run_inputs = {}
+
+    try:
+        last_run_inputs = pickle.load(open("last_run_inputs.pkl", "rb"))
```
Comment (ehoelzl) [inline mlbench_core/cli/cli.py:120]: Ideally, place it under `~/.local/share/mlbench`, using `os.path.join(os.environ['HOME'], ".local", "share", "mlbench")`

File: mlbench_core/cli/cli.py [hunk 3]
```diff
@@ -144,19 +155,24 @@ def run(name, num_workers, gpu, light, dashboard_url):
...
     )
+    current_run_inputs['backend'] = selection
 
     if selection == len(MLBENCH_BACKENDS):
-        backend = click.prompt("Backend", type=str)
+        backend = click.prompt("Backend", type=str, default=last_run_inputs.get('custom_backend', None))
+        current_run_inputs['custom_backend'] = backend
         run_on_all = click.confirm(
-            "Run command on all nodes (otherwise just first node)"
+            "Run command on all nodes (otherwise just first node)", default=last_run_inputs.get('run_on_all', None)
```
Comment (ehoelzl) [inline mlbench_core/cli/cli.py:166]: Not sure if the default value should be `None`. Have you tried it ?

File: mlbench_core/cli/cli.py (hunk 1 above)
Comment (mmilenkoski) [reply mlbench_core/cli/cli.py:121]: I don't think it's necessary to print that no previous inputs were found. In that case, we just offer the existing defaults as we do now.

File: mlbench_core/cli/cli.py (hunk 3 above)
Comment (mmilenkoski) [reply mlbench_core/cli/cli.py:166]: yes, I tested it and passing `None` results in no default value, so the user has to enter the value, just like it is now.

**Pull request merged**